### PR TITLE
Docs updates

### LINKED ISSUE #278: Explain how to use library for simple point arithmetic
Since one of the stated goals of the library is to be a teaching tool for ECC, we should have documentation showing how you can use it to perform some simple operations on the the elliptic points.

Use the simple `y**2 = x**3 + 1x + 4 mod 5` curve in examples (with generator (1, 1))

see also #95

### PR DESCRIPTION
**Pull request by tomato42**
Project description, badge, more updates to docs

Work towards #95
fixes #278

### CODE REVIEW

File: docs/source/basics.rst [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+by specifying the curve using ASN.1 OBJECT IDENTIFIER (OID), which is
+called ``named_curve``. ASN.1 OIDs are a hierarchical system of representing
+types of objects, for example, NIST P-256 curve is identified by the
+1.2.840.10045.3.1.7 OID (in dotted-decimal formatting of the OID, also
+known by the ``prime256v1`` OID node name or short name). Those OIDs
+uniquely, identify a particular curve, but the receiver needs to know
+which numerical OID maps to which curve parameters. Thus the prospect of
+using the explicit encoding, where all the needed parameters are provided
+is tempting, the downside is that curve parameters may specify a *weak*
+curve, which is easy to attack and break (that is deduce the private key
```
Comment (ueno) [inline docs/source/basics.rst:118]: "that is _to_ deduce" maybe?

File: docs/source/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+   ECC
+     Elliptic Curve Cryptography, a term for all the different ways of using
+     elliptic curves in cryptography. Also combined term for :term:`ECDSA`,
+     :term:`EdDSA`, :term:`ECDH`.
+
+   ECDSA
+     Elliptic Curve Digital Signature Algorithm
+
+   EdDSA
+     Edward curve based Digital Signature Algorithm, the alternative
```
Comment (ueno) [inline docs/source/glossary.rst:18]: Edward → Edwards

File: docs/source/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+     digitial signature algorithm that's used for Curve25519 or Curve448
```
Comment (ueno) [inline docs/source/glossary.rst:19]: digitial → digital

File: docs/source/glossary.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+   hybrid
+       A combination of :term:`uncompressed` and :term:`compressed` encodings.
+       Both x and y coordinates are stored just as in :term:`compressed`
+       encoding, but the first byte reflects the sign of the y coordinate. The
+       first byte of the encoding will be equal to 0x06 or 0x7. Hybrid
+       encoding of NIST256p public key will be 65 bytes long.
+
+   PEM
+       The acronym stands for Privacy Enhanced Email, but currently it is used
```
Comment (ueno) [inline docs/source/glossary.rst:60]: Privacy Enhanced _Mail_?

File: docs/source/glossary.rst [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+       deterministically and uniquely into byte strings.
+
+   ASN.1
+       Abstract Syntax Notation 1 is a standard description language for
+       specifying serialisation and deserialisation of data structures in a
+       portable and cross-platform way.
+
+   bytes-like object
+       All the types that implement the buffer protocol. That includes
+       ``str`` (only on python2), ``bytes``, ``bytesarray``, ``array.array``
```
Comment (ueno) [inline docs/source/glossary.rst:79]: ``bytesarray`` → ``bytearray``?

**Review by ueno — APPROVED** [review]
Well written, I could only find minor nits.

File: docs/source/basics.rst (hunk 1 above)
Comment (tomato42) [reply docs/source/basics.rst:118]: yes

File: docs/source/glossary.rst (hunk 2 above)
Comment (tomato42) [reply docs/source/glossary.rst:18]: yes

File: docs/source/glossary.rst (hunk 3 above)
Comment (tomato42) [reply docs/source/glossary.rst:19]: yes

File: docs/source/glossary.rst (hunk 4 above)
Comment (tomato42) [reply docs/source/glossary.rst:60]: yes, should be "Mail"

File: docs/source/glossary.rst (hunk 5 above)
Comment (tomato42) [reply docs/source/glossary.rst:79]: right

**Comment by tomato42** [comment]
@ueno Thanks, I've fixed the issues.

**Pull request merged**